### PR TITLE
Connect to BookStore API

### PR DESCRIPTION
**Pull request by marvin-nyalik**
In this branch:

- I Made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as the backend.
- Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make your HTTP requests to the backend
- I Fetched the books from the Bookstore API
- I Refactored my add book and remove book features to persist changes in the server

Let me know of the required changes

### CODE REVIEW

File: src/components/addBook.js [hunk 1]
```diff
@@ -19,14 +19,15 @@ const AddBook = () => {
...
     event.preventDefault();
     const category = 'Code';
-    dispatch(addBook({
-      title, author, category, item_id: `Item${length + 1}`,
+    await dispatch(addBook({
+      title, author, category, itemId: `Item${length + 1}`,
     }));
     setAuthor('');
     setTitle('');
+    dispatch(fetchBooks());
```
Comment (mckent05) [inline src/components/addBook.js]: **OPTIONAL** - I suggest calling this function in your `addBook` action. Please fix this 🙏. You can check this [article](https://stackoverflow.com/questions/68871562/dispatch-action-on-the-createasyncthunk) for how to dispatch action inside a `createAsyncThunk` function 👍 👍

File: src/components/BookInfoCol.js [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
-  const deleteBook = (id) => {
-    dispatch(removeBook({ item_id: id }));
-  };
 
+  const book = useSelector((state) => Object.entries(state.books.data)
+    .find((book) => bookId === book[0]));
+
+  const deleteABook = async (bookId) => {
+    await dispatch(deleteBook({ bookId }));
+    dispatch(fetchBooks());
```
Comment (mckent05) [inline src/components/BookInfoCol.js]: **OPTIONAL** - I suggest calling this function in your `removeBook` action. Please fix this 🙏. You can check this [article](https://stackoverflow.com/questions/68871562/dispatch-action-on-the-createasyncthunk) for how to dispatch action inside a `createAsyncThunk` function 👍 👍

**Review by mckent05 — APPROVED** [review]
Hi @marvin-nyalik, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job connecting your app to the API 👍 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookInfoCol.js (hunk 2 above)
Comment (marvin-nyalik) [reply src/components/BookInfoCol.js]: I've made the optional changes. Thanks for the heads up

**Pull request merged**